### PR TITLE
adds antagonists to the game

### LINKED ISSUE #11472: Wizards using shapeshift spells to turn into low-health forms instantly die if they have previously taken damage
### Severity

Round Breaking

### Short Description

Any wizards using the shapeshift spell to turn into a mouse or other form with little starting health instantly die if they have taken damage that woud kill that form 

### Reproduction

1. be a wizard
2. get the shapeshift spell/use magic mirror to turn into a vampire
3. suffer somewhat minor damage (10 or 15 shoud suffice,might be as low as 5)
4. use shapeshifting ability
5. die instantly

### Round ID

35327

### Test Merges

_No response_

### BYOND Username

Xomegawarriorx

### Additional information

in the specified round,it happened as i turned into a bat after catching fire for a couple seconds

### LINKED ISSUE #11478: Rev-hud lost midround
## Test Merges: 
Not Supplied / None
## Reproduction:
I was headrev, and at some point in the round, I noticed that I had lost the ability to see who was a rev and who wasn't. I believe this had something to do with my midround obsession.
 Submitted by: fuku (195013019498053633)

### LINKED ISSUE #12950: Mind reading someone who was telepathically communicated to broadcasts ckeys.
### Severity

Exploit

### Short Description

I mind read someone after telepathically communicating with them. Saw ckeysthey in their 

### Reproduction

1.Telepathically comminucate with someone
2.Mind read them
3. see ckeys

### Round ID

38711

### Test Merges

#12922, #12608, #12900, #12919, #12921

### BYOND Username

Noodlecat

### Additional information

_No response_

### LINKED ISSUE #13524: bloodsucker issue
## Test Merges: 
#13273
## Reproduction:
was able to open coffin during daylight and survive but when the coffin was closed again to heal i was stuck asleep

Optional fields:
 Submitted by: wasa656 (542744084536623134)

### LINKED ISSUE #13818: Bloodsuckers can't unbuckle people from their meatspike/rack thing.
### Severity

Inconvenience

### Short Description

Split off from my original issue, bloodsuckers can't seem to manually unbuckle people from the rack they use to thrall people.  This can be annoying if the person in question is mindshielded and therefore cant be enthralled.  Breaking the rack and or waiting for the individual to resist out are the only ways of doing it.

### Reproduction

1. be bloodsucker
2. use spike
3. discover person is mindshielded
4. try to unbuckle them
5. fail

### Round ID

N/A

### Test Merges

_No response_

### BYOND Username

Cark

### Additional information

_No response_

### LINKED ISSUE #13860: Wizard Jaunt is bugged
### Severity

Round Breaking

### Short Description

As wizard, tried to jaunt and it locked my camera on that tile, didn't mist me, and then moved my character back and forth.

### Reproduction

Blinked into sec pod, prepared El Nath, and tried to jaunt.

### Round ID

Round #40662

### Test Merges

_No response_

### BYOND Username

_No response_

### Additional information

_No response_

### LINKED ISSUE #14184: Ability timer stuck reloading
## Test Merges: 
Not Supplied / None
## Reproduction:
Description: After you use an ability such as genetics cryobeam but then click it again to cancel doing it, the reload timer is stuck showing up as 15s. even if you use the ability it doesnt dissapear

 Submitted by: truzkawa (395989780153761794)

### LINKED ISSUE #14453: Ascended predatory lunge for tizimscene bloodsuckers doesnt tear out hearts
## Test Merges: 
Not Supplied / None
## Reproduction:
Use dark embrace, ascended predatory lunge, on a dead body, dont tear out the heart, just aggro grab the body normally

 Submitted by: ImmortalPlagueRat (605167460450304019)

### LINKED ISSUE #14467: pepperspray makes AI HUD unreadable
## Test Merges: 
Not Supplied / None
## Reproduction:
description: when someone is peppersprayed the HUD icons (medical, job, sec) in AI view become pretty much unreadably dark red (unsure if this applies to non-AI HUDs)

 Submitted by: Ryab (260121321948708864)

### LINKED ISSUE #15219: Bloodsucker tasks didn't complete
## Test Merges: 
Not Supplied / None
## Reproduction:
Description: I gain task for 2 hearts. I can't insert hearts into Resting Place, so I insert them into Blood Altar. Blood Altar and Resting Place don't give me option to gain Rank Up, instead offer to eat 1 Rank/550 Blood to gain 1 shadowpoint. Can't gain new tasks either.

 Submitted by: Kojuh (239560674517188609)

### PR DESCRIPTION
**Pull request by tattax**
i'll fill this out later while i test stuff but i need this out to see merge conflicts

EDIT V10 GO:

# Document the changes in your pull request

image

changes how antag huds are handled, makes a new file, organizes everything alphabetically

kills proc holders in exchange for /datum/action/cooldown/spells, changes how devil, hivemind and sling is handled +
everything else basically

changes up some bloodsucker things and add some new sprites, this was supposed to be a bloodsucker PR but we don't have so many things i'd be better to just port things, still ports relevant bloodsucker PRs from fulp that refactor how things work.

ports https://github.com/tgstation/tgstation/pull/67083,
https://github.com/tgstation/tgstation/pull/56958,
https://github.com/tgstation/tgstation/pull/61077,
https://github.com/fulpstation/fulpstation/pull/783,
https://github.com/fulpstation/fulpstation/pull/946,
https://github.com/fulpstation/fulpstation/pull/559,
https://github.com/fulpstation/fulpstation/pull/615,
https://github.com/tgstation/tgstation/pull/61087,
https://github.com/tgstation/tgstation/pull/62718,
https://github.com/fulpstation/fulpstation/pull/884,
https://github.com/tgstation/tgstation/pull/65180,

Closes https://github.com/yogstation13/Yogstation/issues/18572
Closes https://github.com/yogstation13/Yogstation/issues/18382
Closes https://github.com/yogstation13/Yogstation/issues/18467 ?
Closes https://github.com/yogstation13/Yogstation/issues/18345 
Closes https://github.com/yogstation13/Yogstation/issues/18344
Closes https://github.com/yogstation13/Yogstation/issues/18286 ?
Closes https://github.com/yogstation13/Yogstation/issues/18116
Closes https://github.com/yogstation13/Yogstation/issues/18090
Closes https://github.com/yogstation13/Yogstation/issues/18004
Closes https://github.com/yogstation13/Yogstation/issues/17155
Closes https://github.com/yogstation13/Yogstation/issues/16612
Closes https://github.com/yogstation13/Yogstation/issues/16210
Closes https://github.com/yogstation13/Yogstation/issues/15814 ?
Closes https://github.com/yogstation13/Yogstation/issues/15386
Closes https://github.com/yogstation13/Yogstation/issues/15455
Closes https://github.com/yogstation13/Yogstation/issues/15219
Closes https://github.com/yogstation13/Yogstation/issues/11478
Closes https://github.com/yogstation13/Yogstation/issues/11472
Closes https://github.com/yogstation13/Yogstation/issues/12950
Closes https://github.com/yogstation13/Yogstation/issues/13860
Closes https://github.com/yogstation13/Yogstation/issues/13818
Closes https://github.com/yogstation13/Yogstation/issues/13524
Closes https://github.com/yogstation13/Yogstation/issues/14184
Closes https://github.com/yogstation13/Yogstation/issues/14467
Closes https://github.com/yogstation13/Yogstation/issues/14453

# Wiki documentation
  
(most of the changes are reflected on ANTAG TGUI for easy access)

1 - change info on slaughter demon bloodcrawl
2 - add new info on spells (duffel for wiz -> spell that spawns a cursed duffelbag that slows you and bites you, shadow manip for bloodsucker  -> makes different shadow structures based on level)
3 - change info on rank system (now selected on tgui window)
4 - change any info regarding brute damage for lasombra
5 - change info on toreador's objective (now leader) and lasombra's and gangrel's (reduced objective to (1 or 2) times and (1) time respectively)
6 - add info regarding sacrificing on resting place
7 - add info on zombie tzimisce
8 - summon events on 100 dynamic
# Spriting

<details>
  <summary>ANTAG TGUI</summary>

ANTAG INFO
image

NEW ANTAG NUMBER FORMAT
image

BLOODSUCKER
image

image

image

CHANGELING

image

DARKSPAWN

image

WIZAZZ

image

image

</details>

:cl:  MrMelbert, JohnWillard, tralezab, Mothblocks, yours truly, dragomol
rscadd: more mousepointers, more antag UIs
rscadd: adds curse of duffelbag spell
rscadd: adds statues for toreador vampires
rscadd: you can summon events at 100 threat dynamic
rscadd: slaughter demon bloodcrawl spell
rscadd: antag info UI, shows objectives and more!
tweak: tweaks how sunlight, clans and blood is handled for bloodsuckers, adding more dynamics
tweak: changes how some clan objectives worked and starting loadouts
bugfix: fixes a lot of things
imageadd: adds a sprite for mansus inhand and other spells
imageadd: adds sprite for UI interaction
imageadd: changes cult constructs sprites
/:cl:

### CODE REVIEW

**Comment by tattax** [comment]
bold are the travelers that open files changed without further warning

**Comment by Moltijoe** [comment]
wow

**Comment by Cartlord** [comment]
unatomic, wow

**Comment by MajManatee** [comment]
wow who coulda guessed that an unfinished draft PR has merge conflicts. Crazy.

File: code/modules/spells/spell_types/touch/_touch.dm [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+/**
+ * Signal proc for [COMSIG_PARENT_QDELETING] from our attached hand.
+ *
+ * If our hand is deleted for a reason unrelated to our spell,
+ * unlink it (clear refs) and revert the cooldown
+ */
+/datum/action/cooldown/spell/touch/proc/on_hand_deleted(datum/source)
+	SIGNAL_HANDLER
+
+	remove_hand(reset_cooldown_after = TRUE)
```
Comment (JohnFulpWillard) [inline code/modules/spells/spell_types/touch/_touch.dm]:
```suggestion
	INVOKE_ASYNC(src, PROC_REF(remove_hand), null, TRUE)
```

**Review by cuackles — CHANGES_REQUESTED** [review]
- talking sword mob sprite got deleted in mob.dmi, recover it
- deletes show and hide from mob/actions, dont see any replacement to them, either recover them or inform me where the replacement is
- the icon for bite_target is the icon for lilith pact, havent read what the file does but if its not the pointer for lilith pact then change
- need to see how the new lightning icon looks on game 
- need to see high frecuenzy blade onhand ingame

**Review by cuackles — APPROVED** [review]

**Comment by Moltijoe** [comment]
conflicts

**Comment by tattax** [comment]
@Moltijoe

**Comment by TerraTheUnwise** [comment]
yo thanks bro I really wanted antags in this game adds that scare factor. 👍

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
bugs sent on disc for you to fix

**Comment by MajManatee** [comment]
the DMI conflicts in question

**Review by ToasterBiome — APPROVED** [review]

**Comment by DylonBR** [comment]
this is the worst PR to have ever been merged

**Comment by tattax** [comment]
> this is the worst PR to have ever been merged

image

**Pull request merged**